### ISSUE TITLE
Fling from local storage - no cloud

### ISSUE THREAD

**Issue by sagreine** [sagreine/homeGym#3]
right now we go to storage on the cloud and db
can we instead cast local media?
this allows for no-cloud options working
also might just be faster than cloud entirely, so we could put cloud back to if we want to watch it from non-flung mode on TV (e.g. review old videos)

**Comment by sagreine** [comment]
see flutter_local_asses_server, inappwebview

**Comment by sagreine** [comment]
https://github.com/lesnitsky/flutter_local_assets_server/issues/4

**Comment by sagreine** [comment]
just do your own nano clone...

**Comment by sagreine** [comment]
fling a looped version of the 3d animation by default once this is in :)

**Comment by sagreine** [comment]
https://forums.developer.amazon.com/questions/215345/cast-local-media-files-using-fling.html

**Comment by sagreine** [comment]
https://source.codeaurora.org/quic/la//platform/frameworks/av/tree/include/media/mediaplayer.h?id=jb_2.2_rb2_5.2#l111

**Comment by sagreine** [comment]
so this works (obvi needs to be refactored) but if you don't compress it, the video is slooooooooow and it basically doesn't play at all on the firetv because it needs to buffer so much. if you compress, sure, but there is a delay then. hmm. in any event, can probably bring this into the roadmap now, using compression at least.

**Comment by sagreine** [comment]
so seems there is a choice right now:
1) no compression (maybe mess with resolution on recording?), immediate cast but can't really play on my device + wifi
2) compress first. might take up to 90% of the time of the video to do the compression :(. playback is great though

**Comment by sagreine** [comment]
if we're doing this, save to the local gallery too... maybe even set some of the metadata to be friendly

**Comment by sagreine** [comment]
yeck. so we might just have to live with this. at least we can cast locally and make the videos only available to authorized users. maybe have a 'high speed mode' for people or something....
so, build this then.

**Comment by sagreine** [comment]
new plan:
1) super internet mode
2) not that ->
0th start a timer counting down from the rest period, on mobile
first, fling the numbers with a default video (from cloud)
in the background,
second, compress the video
third, fling the compressed video once that is finished
fourth, add premium option for cloud backup and push the video there without waiting for completion

**Comment by sagreine** [comment]
- need to double check port stuff.
- does .mkv file even work?
- implement the rest of the above.

**Comment by sagreine** [comment]
video repeats indefinitely right now.
wouldn't want this in the real world, but good enough for this
that allows it to not stop once the first fling's timer has run out -> but we might not actually care about that, since the timer would ideally be the same / destroyed and recreated?
in any event, for now, we can leave it so it repeats indefinitely. might be preferable anyway if i'm the user

launches server on init, listens, goes from there, more or less works.
next:
zip, fling + cloud at same time = regular mode
super internet mode
cloud as optional mode
cloud permissions to auth - only

**Comment by sagreine** [comment]
.mkv works

need to reorder so it asks you how many reps you got right away, not at the end. otherwise this is basically done. (yay!)
will touch this code as part of 'save local copies of video too' but yea.

super internet mode. might leave that as a far future enhancement

**Comment by sagreine** [comment]
on the TV side, still need to address the "old timer still exists when you cast a new timer at me" piece of this but otherwise ready to go for the most part, so closing.
